syntax = "proto3";

package PROTO;

message UnionInnerOne {
  int32 number = 1;
}

message UnionInnerTwo {
  string str = 1;
}

message UnionInnerThree {
  repeated uint32 values = 1;
}

message UnionContainer {
  oneof msg {
      UnionInnerOne msg1 = 1;
      UnionInnerTwo msg2 = 2;
      UnionInnerThree msg3 = 3;
  }
}